.github/workflows/mergeReleaseBranch.yml

```diff
@@ -0,0 +1,83 @@
+name: 'Merge Release branch'
```

Comment:
This workflow will read the version in main and the provided release branch.  As long as the release branch is `newer` then main it will allow the merge into main. 

The merge into main will trigger the publish workflow to start https://github.com/forcedotcom/salesforcedx-vscode/pull/4606/files#diff-ff0df3629970528f0ac692a38deabc071dc87fdb0112519bba29a20a1023a028

Reply:
Nit: Capitalize Branch; the rest sounds good